File: MovieousDemo/Components/Shared/MDAVPlayerView.m
//
//  MDAVPlayerView.m
//  MovieousDemo
//
//  Created by Chris Wang on 2019/3/10.
//  Copyright © 2019 Movieous Team. All rights reserved.
//

#import "MDAVPlayerView.h"

@implementation MDAVPlayerView

+ (Class)layerClass {
    return AVPlayerLayer.class;
}

+ (instancetype)playerViewWithPlayer:(AVPlayer *)player {
    return [[self alloc] initWithPlayer:player];
}

- (instancetype)initWithPlayer:(AVPlayer *)player {
    if (self = [super init]) {
        ((AVPlayerLayer *)self.layer).player = player;
    }
    return self;
}

- (AVLayerVideoGravity)videoGravity {
    return ((AVPlayerLayer *)self.layer).videoGravity;
}

- (void)setVideoGravity:(AVLayerVideoGravity)videoGravity {
    ((AVPlayerLayer *)self.layer).videoGravity = videoGravity;
}

@end

File: MovieousDemo/Components/Audience/MDAVPlayerView.swift
//
//  MDAVPlayerView.swift
//  MovieousDemo
//
//  Created by Chris Wang on 2019/4/10.
//  Copyright © 2019 Movieous Team. All rights reserved.
//

import UIKit
import AVFoundation

class MDAVPlayerView: UIView {
    override class var layerClass: AnyClass {
        get {
            return AVPlayerLayer.self
        }
    }
    
    var videoGravity: AVLayerVideoGravity {
        set {
            (self.layer as! AVPlayerLayer).videoGravity = newValue
        }
        get {
            return (self.layer as! AVPlayerLayer).videoGravity
        }
    }
    
    var player: AVPlayer! {
        didSet {
            (self.layer as! AVPlayerLayer).player = self.player
        }
    }
}
